api/pkg/apis/v1alpha1/managers/activations/activations-manager.go

```diff
@@ -215,8 +218,8 @@ func (t *ActivationsManager) ReportStatus(ctx context.Context, name string, name
 	var err error = nil
 	defer observ_utils.CloseSpanWithError(span, &err)
 	defer observ_utils.EmitUserDiagnosticsLogs(ctx, &err)
-	lock.Lock()
-	defer lock.Unlock()
+	t.KeyLockProvider.Lock(name + namespace)
```

Comment:
name + namespace is not a good key. 
What if name is "a", but namespace is "bc", it will share the same key as the object name is "ab" and namespace is "c"

Reply:
Write a uniform key generator in util functions with special separator.

Reply:
Is this keylockprovider a singleton? if so, different resource may have the same key

Reply:
@msftcoderdjw I think the collision is not a big problem. It only caused objects with same key share the same lock. But I am OK to add a separator if you insist.
@iwangjintian Each keylockprovider maintains its own lockMap so different resource manager will load their own lockMap.

Reply:
@iwangjintian 's concern is valid, if the keylockprovider is not singleton, what if two manager has its own lockMap, there's no lock between these two managers. 
My feeling is better to use singleton lockMap, and the key is "namespace"+"sep"+"resourceType"+"sep"+""resourceName".